src/util.rs

```diff
@@ -37,7 +39,9 @@ pub fn create_helper<F, R>(
 where
     F: Fn(PathBuf) -> io::Result<R>,
 {
-    for _ in 0..::NUM_RETRIES {
+    let num_retries = if random_len != 0 { ::NUM_RETRIES } else { 1 };
```

Comment:
While testing this I was playing with attempting to create a file I could guarantee already existed by setting `random_len` to `0`. That's a bit of a footgun because you'll effectively try create the file 2147483648 times before giving up. Setting `random_len` to `0` is pretty silly, but we might want to just panic or use a much smaller number of retries if the available entropy is too small. I thought I'd leave this here so we could think about it more.

Reply:
One try sounds reasonable, nice catch! I can see users wanting to *explicitly* specify the path of a temporary file (if they just want a file that's automatically cleaned up).

Really, we should probably set NUM_RETRIES to something sane like 256 but we can do that later.